### PR TITLE
Add log verbosity option to test_runner

### PR DESCRIPTION
**Pull request by wilzbach**
> In order to not do the link step, the -c switch needs to be thrown on the compiler. I don't see that in the diff - can you check, please?

I think allowing to log the executed commands might be useful ;-)

Hence [the output](http://sprunge.us/cMLI) with `DMD_TEST_COVERAGE=1 DMD_TEST_VERBOSITY=1` looks shows `-c`.

``
cat log | wc -l                              # prints 862
cat mylog | grep "-c" | wc -l     # prints 862

```

I tried to keep this option as KISS as possible, but removed the automatic string concatenation at CTFE in the help message because AFAIK that's a deprecated feature?
```

### CODE REVIEW

**Comment by wilzbach** [comment]
> No coverage report found for master at 51c2626.

That's was a problem on the CodeCov side as their first patch for mapping the files correctly was buggy. As you can see they already fixed it and future commits should be processed/annotated again.

File: test/d_do_test.d [hunk 1]
```diff
@@ -16,26 +16,29 @@ import core.sys.posix.sys.wait;
...
+          "      OS:                 win32, win64, linux, freebsd, osx\n",
+          "      RESULTS_DIR:        base directory for test results\n",
+          "      DMD_TEST_COVERAGE   omits the linking and running of all tests\n",
+          "      DMD_TEST_VERBOSITY  set the script log verbosity (default 0)\n",
+          "\n",
+          "   windows vs non-windows portability env vars:\n",
+          "      DSEP:               \\\\ or /\n",
+          "      SEP:                \\ or /\n",
+          "      OBJ:                .obj or .o\n",
+          "      EXE:                .exe or <null>\n");
```
Comment (MartinNowak) [inline test/d_do_test.d]: Should be a heredoc string, but not part of this PR.

**Comment by MartinNowak** [comment]
I'd rather just export [`QUIET=@`](https://github.com/dlang/dmd/blob/3476429c3daa5ef2be81108e6446ce0955fe5f98/test/Makefile#L90) and interpret that as verbose on/off.
Don't think we'll need a more complex log level anytime soon.

**Comment by wilzbach** [comment]
> I'd rather just export QUIET=@ and interpret that as verbose on/off.
> Don't think we'll need a more complex log level anytime soon.

Ok I didn't know how to check whether `QUIET` is already set in a cross-platform make syntax, so it just checks for `0`. It's not that nice as the only way to get the verbose output is `QUIET=0 make ...` , but I assume this won't be used much anyways.

**Comment by andralex** [comment]
I like `QUIET=@` too. Should we close this?

**Comment by wilzbach** [comment]
> I like QUIET=@ too. Should we close this?

The idea here was to print the exact commands that are executed by the test_runner (in this case it was to show Walter that `-c` is used), so I thought that might be interesting in the future too.
However there is nothing fancy about this PR and I don't depend on it at by any means.

**Comment by wilzbach** [comment]
I just close this, it's trivial, I don't need it and no one seemed interested.
